video_worker/tests/test_video_worker.py

```diff
@@ -0,0 +1,535 @@
+            }
+        ),
+        (
+            {
+                'encode_profile': 'static-pipeline',
+                'path_exists': False
+            }
+        ),
+        # Success
+        (
+            {
+                'encode_profile': 'hls'
+            }
+        ),
+        (
+            {
+                'encode_profile': 'static-pipeline'
+            }
+        )
+    )
+    @patch.object(VideoWorker, '_static_pipeline')
+    @patch.object(VideoWorker, '_hls_pipeline')
+    @patch.object(VideoWorker, '_update_api')
+    @patch('video_worker.video_images.VideoImages.settings_setup')
+    @patch('video_worker.video_images.VideoImages.create_and_update')
+    @patch.object(deliverable_route, 'apply_async')
+    @patch.object(video_worker_logger, 'error')
+    @patch('shutil.rmtree')
+    @patch('os.path.exists')
+    @patch('os.mkdir')
+    def test_run(self, mock_data, mock_mkdir, mock_exists, mock_rmtree, mock_logger, celeryapp_sync_mock,
+                 vide_images_create_and_update_mock, video_images_setup_mock, mock_update_api, mock_hls_pipeline,
+                 mock_static_pipeline):
+        """
+        Test that `run` method works correctly.
+        """
+        is_valid = mock_data.get('is_valid', True)
+        is_valid_engine_intake = mock_data.get('is_valid_engine_intake', True)
+        error_message = mock_data.get('error_message', '')
+        encoded_profile = mock_data.get('encode_profile', None)
+        path_exists = mock_data.get('path_exists', True)
+
+        self.VW.instance_yaml = TEST_INSTANCE_YAML
+        self.VW.settings = WS.settings_dict
+        self.VW.encode_profile = encoded_profile
+        self.VW.VideoObject.valid = is_valid
+        self.VW.jobid = 'dummy-jobid'
+        self.VW.endpoint_url = '/dummy-endpoint-url'
+
+        # First 2 calls to os.path.exists will be called in WS.run() so we need to make sure our TEST_INSTANCE_YAML
+        # file path is check correctly.
+        mock_exists.side_effect = [True, True, path_exists]
+
+        video_images_setup_mock.return_value = WS.settings_dict
+
+        def change_video_valid(self):
+            """
+            Changes Video.valid to True when validate() is called.
+            """
+            self.valid = True
+            self.val_id = 'dummy-val-id'
+            self.veda_id = 'dummy-veda-id'
+
+        def change_video_invalid(self):
+            """
+            Changes Video.valid to False when validate() is called.
+            """
+            self.valid = False
+            self.val_id = 'dummy-val-id'
+            self.veda_id = 'dummy-veda-id'
+
+        def change_video_valid_intake(self):
+            """
+            Changes VM.VideoObject.valid to False.
+            """
+            self.VideoObject.valid = True
+
+        def change_video_invalid_intake(self):
+            """
+            Changes VM.VideoObject.valid to True.
+            """
+            self.VideoObject.valid = False
+
```

Comment:
I couldn't find an alternate way. for each mock to change values dynamically we needed a method and that doesn't take other parameters so I had to chose this approach, happy to listen to other suggestions